Sources/Layers/Concat.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Concat.swift
+//  Bender
+//
+//  Created by Diego Ernst on 6/27/17.
+//
+//
+
+import MetalPerformanceShaders
+
+open class Concat: NetworkLayer {
+
+    var pipeline: MTLComputePipelineState!
+    let axis: LayerSizeAxis
+
+    let maxInputTextures = 10 // must match kernel max input textures
+
+    public init(axis: LayerSizeAxis, id: String? = nil) {
+        self.axis = axis
+        super.init(id: id)
+    }
+
+    open override func initialize(network: Network, device: MTLDevice) {
+        super.initialize(network: network, device: device)
+        let incoming = getIncoming()
+
+        // correctness checks
+
+        // TODO: implement shader to support concat along z without restricting it to multiples of 4
+
+        assert(axis != .f || incoming.reduce(true) { $0.0 && ($0.1.outputSize.f % 4 == 0) }, "Concat: all z dimensions must be multiple of 4")
+        assert(!incoming.isEmpty, "Concat: expects at least one inputs")
```

Comment:
Maybe you can move this up, if this check fails then the previous one should be needed. I think it can help to understand the checks done, I would first think to check if the sequence is empty and then the other checks. Just a minor comment, nothing to worry about